### ISSUE TITLE
Indicator Volume by Price not working correctly in relative ordinate

### ISSUE THREAD

**Issue by robocop98** [highcharts/highcharts#16277]
#### Expected behaviour
Points of volume-by-price indicator should be drawn with relative values when using compare: 'percent'.

When switching between absolute and relative ordinate the volume-by-price bars should remain on the same relative position to its underlying series line, like simple indicators (SMA, ...) and multi-line indicator (Bollinger Bands) do.

#### Actual behaviour
volume-by-price is always drawn with absolute values, so not shown if switching to relative ordinates (compare: 'percent'),

Exception: indicator can be seen randomly, if its underlying series has values near the percentual ones.

EDIT: vbp Indicator is also not working on logarithmic ordinates, but not sure if it makes sense to use it there.

#### Live demo with steps to reproduce
Absolute ordinate works: https://jsfiddle.net/46d28qxb/1/
Relative ordinate does not work, volume-by-price indicator is outside of the view: https://jsfiddle.net/46d28qxb/

Difference between both fiddles:
- at main series and vbp series: compare: 'percent'
- at vbp series: compareToMain: true

#### Product version
Highstocks 9.2.0 - 9.2.2

#### Affected browser(s)
Tested in Chrome and Firefox, but issue should not depend on browser.

Thank you for your constant support @ Highcharts team

**Comment by raf18seb** [comment]
Hi @robocop98, thanks for creating the ticket!

The VBP indicator is slightly different from the others - it's not a simple line, the values are not distributed along the x-axis and the standard `compare` feature wasn't developed to cover this case.
How would you like to compare the VBP columns values (green and red)? Could you describe a simple case?

**Comment by robocop98** [comment]
Hi @raf18seb , thanks for picking up this topic.

The use cases we currently have in our setup:
- in a stock chart with one instrument series we can manually switch the y-axis between absolute and relative coordinates
-  either see the

**Comment by raf18seb** [comment]
@robocop98, thanks for clarifying!

> The indicators y-coordinates are correct in absolute view, but would need re-calculation for the relative view (compare='percent'), so that it is shown at "same location" like in absolute view.

Yes, this makes sense. In other words, the candlestick series would be calculated in the `compare` mode and the VbP indicator would be translated to the correct place, so it fits new y-axis extremes.

However, could you clarify the options you are describing? Especially `compareToMain`.
Did you mean that:
- if we set `compareToMain: true` on the VbP series, then the VbP would be calculated on the main candlestick values, so the VbP would stay the same (as it works now), and
- if we set `compareToMain: false` on the VbP series, then the VbP would be calculated on the new candlestick `compare` values so it would be translated to the relative view

Right?

**Comment by robocop98** [comment]
Hi @raf18seb, thank you for progressing on this topic.

> In other words, the candlestick series would be calculated in the compare mode and the VbP indicator would be translated to the correct place, so it fits new y-axis extremes.

Yes exactly.

Regarding the `compareToMain` option:
After re-reading the documentation https://api.highcharts.com/highstock/series.vbp.compareToMain I am not really sure what a setting of `false` accomplishes on this option. 

We always use `compareToMain: true`, so that the indicators are compared against their (linked to) main series.
And at least with `compareToMain: true` the VbP series should be translated to the relative view using the values of the main series, if `compare: 'percent'` has been set.

Unfortunately I do not really know what should happen, if `compareToMain: false` is set,  sorry.

Do you want to know more details on other "compare-ish" options we might use?

**Comment by karolkolodziej** [comment]
_Internal note:_
Might be similar to: https://github.com/highcharts/highcharts/pull/16075

**Comment by robocop98** [comment]
Dear Highcharts devs, 

When retesting the issue with https://jsfiddle.net/46d28qxb/ the VbP indicator still seems to be broken in compare mode.
As far as I understand, the issue should be fixed in v9.3.1 of Highcharts / Highstocks. Or will it the fix be released later?

**Comment by raf18seb** [comment]
Hi @robocop98!

It is available in the current version (9.3.1).

> if we set compareToMain: true on the VbP series, then the VbP would be calculated on the main candlestick values, so the VbP would stay the same (as it works now), and
if we set compareToMain: false on the VbP series, then the VbP would be calculated on the main candlestick values, but would be translated to the relative view (compare values)

As we discussed, have you tried setting `compareToMain: false`?
`compareToMain: true` demo: https://jsfiddle.net/BlackLabel/q0k4afev/
`compareToMain: false` demo: https://jsfiddle.net/BlackLabel/qsgd95kL/
